src/config.coffee

```diff
@@ -241,8 +241,9 @@ module.exports = class Config extends EventEmitter
 		deviceRegister.generateUniqueKey()
 
 	generateRequiredFields: =>
-		@getMany([ 'uuid', 'deviceApiKey', 'apiSecret', 'logsChannelSecret' ])
-		.then ({ uuid, deviceApiKey, apiSecret, logsChannelSecret }) =>
+		@getMany([ 'uuid', 'deviceApiKey', 'apiSecret', 'logsChannelSecret', 'offlineMode' ])
+		.then ({ uuid, deviceApiKey, apiSecret, logsChannelSecret, offlineMode }) =>
+			return if offlineMode
```

Comment:
@CameronDiver shouldn't we still generate a uuid if it doesn't exist? e.g. for preloaded apps, since they have a RESIN_DEVICE_UUID env var